crates/acvm/src/lib.rs

```diff
@@ -33,7 +42,7 @@ pub trait PartialWitnessGenerator {
         &self,
         initial_witness: &mut BTreeMap<Witness, FieldElement>,
         gates: Vec<Gate>,
-    ) -> Result<(), OPCODE> {
+    ) -> Result<(), GateResolution> {
```

Comment:
I don't think all of GateResolution should necessarily be part of the Err variant here. Some GateResolution variants like Resolved or possibly Skip are not errors and wouldn't make sense to early-return as an error.
Alternatively, since GateResolution already has the positive and negative return values, we should just return it directly instead of wrapping it in a Result.

Reply:
+1 for this note. The only case previously where we were returning an error early was an unsupported opcode when solving a gadget, which is now covered as a return value inside `GateResolution`.  Then when we call the PWG `solve` we can just directly match on the enum values rather than the Err side of a result.

Reply:
Done